app/screens/flagged_posts/flagged_posts.js

```diff
@@ -0,0 +1,314 @@
+    };
+
+    handleClosePermalink = () => {
+        const {actions} = this.props;
+        actions.selectFocusedPostId('');
+        this.showingPermalink = false;
+    };
+
+    handlePermalinkPress = (postId, teamName) => {
+        this.props.actions.loadChannelsByTeamName(teamName);
+        this.showPermalinkView(postId, true);
+    };
+
+    keyExtractor = (item) => item;
+
+    onNavigatorEvent = (event) => {
+        if (event.type === 'NavBarButtonPress') {
+            if (event.id === 'close-settings') {
+                this.props.navigator.dismissModal({
+                    animationType: 'slide-down',
+                });
+            }
+        }
+    };
+
+    previewPost = (post) => {
+        Keyboard.dismiss();
+
+        this.showPermalinkView(post.id, false);
+    };
+
+    renderDateHeader = (date, index) => {
+        return (
+            <DateHeader
+                date={date}
+                index={index}
+            />
+        );
+    };
+
+    renderEmpty = () => {
+        const {formatMessage} = this.context.intl;
+        const {theme} = this.props;
+        const icon = (
+            <IonIcon
+                size={76}
+                color={changeOpacity(theme.centerChannelColor, 0.4)}
+                name='ios-flag-outline'
+            />
+        );
+
+        return (
+            <NoResults
+                description={formatMessage({
+                    id: 'mobile.flagged_posts.empty_description',
+                    defaultMessage: 'Flags are a way to mark messages for follow up. Your flags are personal, and cannot be seen by other users.',
+                })}
+                icon={icon}
+                title={formatMessage({id: 'mobile.flagged_posts.empty_title', defaultMessage: 'No Flagged Posts'})}
+                theme={theme}
+            />
+        );
+    };
+
+    renderPost = ({item, index}) => {
+        const {postIds} = this.props;
+        const {managedConfig} = this.state;
+        if (item.indexOf(DATE_LINE) === 0) {
+            const date = item.substring(DATE_LINE.length);
+            return this.renderDateHeader(new Date(date), index);
+        }
+
+        let separator;
+        const nextPost = postIds[index + 1];
+        if (nextPost && nextPost.indexOf(DATE_LINE) === -1) {
+            separator = this.renderPostSeparator();
+        }
+
+        return (
+            <View>
+                <ChannelDisplayName postId={item}/>
+                <SearchResultPost
+                    postId={item}
+                    previewPost={this.previewPost}
+                    goToThread={this.goToThread}
+                    navigator={this.props.navigator}
+                    onPermalinkPress={this.handlePermalinkPress}
+                    managedConfig={managedConfig}
+                    showFullDate={false}
+                />
+                {separator}
+            </View>
+        );
+    };
+
+    renderPostSeparator = () => {
```

Comment:
Though it's convenient to have this per parent component, I feel like this should be pulled out into separate smaller component. I can see places where we can reuse this. Those that can be pulled out are:
- renderDateHeader
- renderEmpty
- renderPost
- renderPostSeparator